core/src/test/java/org/infinispan/distribution/ZeroCapacityAdministrationTest.java

```diff
@@ -47,6 +55,30 @@ public void testDefineClusterConfiguration() {
       assertNotNull(node1.getCacheConfiguration("zero-template"));
    }
 
+   public void testCreateNewClusteredCacheFromZeroToRemote() {
+      Configuration config = new ConfigurationBuilder()
+            .clustering().cacheMode(CacheMode.DIST_SYNC)
+            .build();
+
+      // The join command is sent to the coordinator.
+      assertThat(node1.isCoordinator()).isTrue();
+
+      // Make sure this will trigger a remote call to create the cache.
+      int tries = 0;
+      String cacheName = "another-cache";
+      ScopedState ss = new ScopedState(CACHE_SCOPE, cacheName);
+      while (!DistributionTestHelper.isFirstOwner(node1.getCache(CONFIG_STATE_CACHE_NAME), ss)) {
+         if (tries > 50) fail("Exceeded attempts to find configuration mapping to remote");
```

Comment:
I like the simplicity of this. I had originally thought we would use mocks etc to ensure that the firstOwner is node1, but maybe this is good enough.

One thing I have noticed though, is that if the test fails (due to reverting your fix) it's unable to cleanup it's state and must be interrupted.

Reply:
The reason is that during the cleanup, each cache is stopped individually. Since we have the cache waiting for the non-zero capacity node to join, the cleanup hangs as the cache future [1] is never completed. 

[1]
 https://github.com/infinispan/infinispan/blob/eb7e866fa6a8ce2219483d5d9c5fdb99622dbcf7/core/src/main/java/org/infinispan/manager/DefaultCacheManager.java#L550-L553

I've updated the test to avoid that. If we have a regression, it will be easier to identify.